src/polyfem/solver/SparseNewtonDescentSolver.hpp

```diff
@@ -18,7 +18,7 @@ namespace cppoptlib
 		using typename Superclass::Scalar;
 		using typename Superclass::TVector;
 
-		SparseNewtonDescentSolver(const json &solver_params, const json &linear_solver_params);
+		SparseNewtonDescentSolver(const json &solver_params, const json &linear_solver_params, const double dt = 1.0);
```

Comment:
do we really want the default here?

Reply:
Just left it for backwards compatibility. It is only used to divide xDelta, so a default of one produces the same old behavior.

Reply:
yep, I was worried that you missed one dt. I think we can always pass it (and pass 1 or -1 for non-time-dependent). The old behavior uses grad norm that's unaffected